editor/plugins/animation_library_editor.cpp

```diff
@@ -0,0 +1,689 @@
+		undo_redo->add_do_method(this, "_update_editor", player);
+		undo_redo->add_undo_method(this, "_update_editor", player);
+		undo_redo->commit_action();
+	}
+}
+
+void AnimationLibraryEditor::_load_library() {
+	List<String> extensions;
+	ResourceLoader::get_recognized_extensions_for_type("AnimationLibrary", &extensions);
+
+	file_dialog->set_title("Load Animation");
+	file_dialog->clear_filters();
+	for (const String &K : extensions) {
+		file_dialog->add_filter("*." + K);
+	}
+
+	file_dialog->set_file_mode(EditorFileDialog::FILE_MODE_OPEN_FILE);
+	file_dialog->set_current_file("");
+	file_dialog->popup_centered_ratio();
+
+	file_dialog_action = FILE_DIALOG_ACTION_OPEN_LIBRARY;
+}
+
+void AnimationLibraryEditor::_file_popup_selected(int p_id) {
+	Ref<AnimationLibrary> al = player->call("get_animation_library", file_dialog_library);
+	Ref<Animation> anim;
+	if (file_dialog_animation != StringName()) {
+		anim = al->get_animation(file_dialog_animation);
+		ERR_FAIL_COND(anim.is_null());
+	}
+	switch (p_id) {
+		case FILE_MENU_SAVE_LIBRARY: {
+			if (al->get_path().is_resource_file()) {
+				EditorNode::get_singleton()->save_resource(al);
+				break;
+			}
+		}
+			[[fallthrough]];
+		case FILE_MENU_SAVE_AS_LIBRARY: {
+			file_dialog->set_file_mode(EditorFileDialog::FILE_MODE_SAVE_FILE);
+			file_dialog->set_title(TTR("Save Library"));
+			if (al->get_path().is_resource_file()) {
+				file_dialog->set_current_path(al->get_path());
+			} else {
+				file_dialog->set_current_file(String(file_dialog_library) + ".res");
+			}
+			file_dialog->clear_filters();
+			List<String> exts;
+			ResourceLoader::get_recognized_extensions_for_type("AnimationLibrary", &exts);
+			for (const String &K : exts) {
+				file_dialog->add_filter("*." + K);
+			}
+
+			file_dialog->popup_centered_ratio();
+			file_dialog_action = FILE_DIALOG_ACTION_SAVE_LIBRARY;
+		} break;
+		case FILE_MENU_MAKE_LIBRARY_UNIQUE: {
+			StringName lib_name = file_dialog_library;
+
+			Ref<AnimationLibrary> ald = al->duplicate();
+
+			UndoRedo *undo_redo = EditorNode::get_singleton()->get_undo_redo();
+			undo_redo->create_action(vformat(TTR("Make Animation Library Unique: %s"), lib_name));
+			undo_redo->add_do_method(player, "remove_animation_library", lib_name);
+			undo_redo->add_do_method(player, "add_animation_library", lib_name, ald);
+			undo_redo->add_undo_method(player, "remove_animation_library", lib_name);
+			undo_redo->add_undo_method(player, "add_animation_library", lib_name, al);
+			undo_redo->add_do_method(this, "_update_editor", player);
+			undo_redo->add_undo_method(this, "_update_editor", player);
+			undo_redo->commit_action();
+
+		} break;
+		case FILE_MENU_EDIT_LIBRARY: {
+			EditorNode::get_singleton()->push_item(al.ptr());
+		} break;
+
+		case FILE_MENU_SAVE_ANIMATION: {
+			if (anim->get_path().is_resource_file()) {
+				EditorNode::get_singleton()->save_resource(anim);
+				break;
+			}
+		};
+			[[fallthrough]];
```

Comment:
Also removed redundant semicolon.
```suggestion
		case FILE_MENU_SAVE_ANIMATION: {
			if (anim->get_path().is_resource_file()) {
				EditorNode::get_singleton()->save_resource(anim);
				break;
			}
			[[fallthrough]];
		}
```